databroker/tests/test_v2/generic.py

```diff
@@ -28,26 +28,46 @@ def compare(a, b):
     a_indexed = {}
     b_indexed = {}
     for name, doc in a:
+        if name == 'resource':
+            # Check for an extraneous duplicate key in old documents.
+            if 'id' in doc:
+                assert doc['id'] == doc['uid']
+                doc = doc.copy()
+                doc.pop('id')
         if name == 'datum':
```

Comment:
I think this should be an elif

Reply:
Good catch. Pushed.

Reply:
Oops, actually it was intentionally an `if`. First we handle the special case of `resource`; then we go on to handle the various ways that documents contain a unique ID (or not). We want `resource` to hit the `else` path of that second part so that it is added to `a_indexed`.

Reply:
Force-pushed to remove commit 305182ab688ecd8ad8e997f2f997a5ba174c0426 and revert to `if`.